packages/orama/src/trees/avl.ts

```diff
@@ -198,121 +237,161 @@ export function create<K, V>(key: K, value: V): Node<K, V> {
   }
 }
 
-export function insert<K, V>(node: Node<K, V>, key: K, value: V): Node<K, V> {
-  if (!node) {
-    return create(key, value)
+export function insert<K, V>(root: Node<K, V>, key: K, value: V): Node<K, V> {
+  let parent = null;
+  let current = root;
+
+  while (current !== null) {
+    parent = current;
+    if (key < current.key) {
+        current = current.left as Node<K, V>;
+    } else if (key > current.key) {
+        current = current.right as Node<K, V>;
+    } else {
+      // assuming value is an array here
+      (current.value as string[]) = (current.value as string[]).concat(value as string);
```

Comment:
Array concat is very slow compared to push, might be worth investigating

Reply:
There is an excellent article to talk about differences of spread, concat, push, etc... that can bring more ideas: https://jpcamara.com/2023/03/07/making-tanstack-table.html

The general idea is: if `current.value` can mutate, then `.push` is the best option, otherwise, `concat` probably will be faster if you don't use a library like `immer`.